flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionManager.java

```diff
@@ -104,12 +192,40 @@ public void shutdown() {
 
             registeredPartitions.clear();
 
+            requestPartitionNotifiers.clear();
+
             isShutdown = true;
 
             LOG.debug("Successful shutdown.");
         }
     }
 
+    /**
+     * Check whether the partition notifier is timeout.
+     */
+    private void checkRequestPartitionNotifiers() {
+        List<PartitionRequestNotifier> timeoutPartitionRequestNotifiers = new LinkedList<>();
+        synchronized (registeredPartitions) {
+            if (isShutdown) {
+                return;
+            }
+            long now = System.currentTimeMillis();
+            Iterator<Map.Entry<ResultPartitionID, InputRequestNotifierManager>> iterator = requestPartitionNotifiers.entrySet().iterator();
+            while (iterator.hasNext()) {
+                Map.Entry<ResultPartitionID, InputRequestNotifierManager> entry = iterator.next();
+                InputRequestNotifierManager partitionRequestNotifiers = entry.getValue();
+                partitionRequestNotifiers.removeExpiration(now, partitionNotifierTimeout.toMillis(), timeoutPartitionRequestNotifiers);
+                if (partitionRequestNotifiers.isEmpty()) {
+                    iterator.remove();
+                }
+            }
+        }
+        for (PartitionRequestNotifier partitionRequestNotifier : timeoutPartitionRequestNotifiers) {
+            partitionRequestNotifier.notifyPartitionRequestTimeout();
+        }
+        partitionNotifierTimeoutChecker.schedule(this::checkRequestPartitionNotifiers, partitionNotifierTimeout.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
`partitionNotifierTimeoutChecker` has not been closed anywhere.

Reply:
Done, use rpc scheduled executor instead